hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azure/AzureNativeFileSystemStore.java

```diff
@@ -2761,18 +2816,23 @@ public boolean delete(String key) throws IOException {
 
   @Override
   public void rename(String srcKey, String dstKey) throws IOException {
-    rename(srcKey, dstKey, false, null, true);
+    rename(srcKey, dstKey, false, null, true, null);
+  }
+
+  @Override
+  public void rename(String srcKey, String dstKey, String eTag) throws IOException {
+    rename(srcKey, dstKey, false, null, true, eTag);
```

Comment:
lets define what etag is this. is this etag on src or dst.

Reply:
Done